### ISSUE TITLE
Add features to KFold cross-validation

### ISSUE THREAD

**Issue by Mec-iS** [smartcorelib/smartcore#10]
Folllowing [sklearn interface](https://github.com/scikit-learn/scikit-learn/blob/0864c5896804bc1066c02bcb1443c962cabdc420/sklearn/model_selection/_split.py#L109):
* other KFold implementations: 'GroupKFold', 'LeaveOneGroupOut', 'LeaveOneOut', 'LeavePGroupsOut', 'LeavePOut', etc.
* consider usage of `derive-builder` for defaults
* ...

**Comment by Mec-iS** [comment]
@VolodymyrOrlov please help with prioritising the features

**Comment by VolodymyrOrlov** [comment]
@Mec-iS the goal is to have `cross_val_score` function somewhere in the `model_selection` module that evaluates performance of a model by training and running an estimator on k folds of data. 

In term of parameters, at minimum we should support:

* `estimator`
* `X`
* `y`
* `scoring`
* `cv`

Ideally we should also evaluate different folds in parallel (`n_jobs` parameter), but sequential evaluation in v0.2.0 will do as well. 

To achieve this goal we need to have:
- a trait for an estimator(s). Right now non of the algorithms we have implement a common trait
- a trait for a scoring metric. Again, we do not have a common trait for metrics from the `metrics` module
- at least one implementation for `cv` (we have it)

So the next step would be to start experimenting with traits for the estimator and metrics

**Comment by Mec-iS** [comment]
Subtasks:
* move `KFold` functionalities in its own module out of `mod.rs`
* add `KFold` usage examples in documentation header

**Comment by VolodymyrOrlov** [comment]
one more suggestion for subtask. We need a new method in BaseMatrix and BaseVector to select rows of the matrix that belong to given indices, similar to [Numpy's take](https://numpy.org/doc/stable/reference/generated/numpy.take.html).
We'll use this method in `cross_val_score ` to take subset of rows that belong to a fold

**Comment by VolodymyrOrlov** [comment]
I was looking for a good visualization of CV that we can use for the first page of smartcorelib.org and found [this plot in Scikit learn ](https://scikit-learn.org/stable/auto_examples/model_selection/plot_cv_predict.html#sphx-glr-auto-examples-model-selection-plot-cv-predict-py). I think we can do something similar with `plotters` crate but we'll need `cross_val_score` function.
